plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperation.java

```diff
@@ -0,0 +1,114 @@
+package com.google.cloud.tools.eclipse.appengine.localserver.server;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.IServer;
+import org.eclipse.wst.server.core.model.IModuleResource;
+import org.eclipse.wst.server.core.model.IModuleResourceDelta;
+import org.eclipse.wst.server.core.model.PublishOperation;
+import org.eclipse.wst.server.core.model.ServerBehaviourDelegate;
+import org.eclipse.wst.server.core.util.PublishHelper;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Handles the publishing operations for the App Engine development server.
+ */
+public class LocalAppEnginePublishOperation extends PublishOperation {
+  private static final String PLUGIN_ID = LocalAppEnginePublishOperation.class.getName();
+
+  /**
+   * @throws {@link CoreException} if status list is not empty
+   */
+  private static void checkStatuses(List<IStatus> statusList) throws CoreException {
```

Comment:
Could this method be called something more expressive? `failOnError` or similar?